x-pack/plugins/security_solution/public/common/components/event_details/insights/insights.tsx

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { EuiFlexGroup, EuiFlexItem, EuiTitle } from '@elastic/eui';
+import { find } from 'lodash/fp';
+
+import * as i18n from './translations';
+
+import type { BrowserFields } from '../../../containers/source';
+import type { TimelineEventsDetailsItem } from '../../../../../common/search_strategy/timeline';
+import { useGetUserCasesPermissions } from '../../../lib/kibana';
+import { useIsExperimentalFeatureEnabled } from '../../../hooks/use_experimental_features';
+import { RelatedAlertsByProcessAncestry } from './related_alerts_by_process_ancestry';
+import { RelatedCases } from './related_cases';
+import { RelatedAlertsBySourceEvent } from './related_alerts_by_source_event';
+import { RelatedAlertsBySession } from './related_alerts_by_session';
+
+interface Props {
+  browserFields: BrowserFields;
+  eventId: string;
+  data: TimelineEventsDetailsItem[];
+  timelineId: string;
+  isReadOnly?: boolean;
+}
+
+/**
+ * Displays several key insights for the associated alert.
+ */
+export const Insights = React.memo<Props>(
+  ({ browserFields, eventId, data, isReadOnly, timelineId }) => {
+    const isRelatedAlertsByProcessAncestryEnabled = useIsExperimentalFeatureEnabled(
+      'insightsRelatedAlertsByProcessAncestry'
+    );
+    const processEntityField = find({ category: 'process', field: 'process.entity_id' }, data);
```

Comment:
more of an organizational thing, but I think it would be good in the long run to make each insight it's own file. So like cases-insights.tsx could be something like this: It'll make it easier in the long run to reason about I think and make it easier to add additional insights and re-use them.

```
export const useCasesInsights = () => { ... return canShowCasesInsights }

export const CasesInsight = ({ canShowCasesInsights, eventId }) => {
 return canShowCasesInsights ? (
            <EuiFlexItem>
              <RelatedCases eventId={eventId} />
            </EuiFlexItem>
   )
   : null;
}
```

Reply:
Yeah agreed, not super happy about the rendering duplication :(